### ISSUE TITLE
Evaluation order bug?

### ISSUE THREAD

**Issue by bakul** [kevinlawler/kona#538]
This used to work in kona (tested on a 6 year old version).... Also works in k3.
````
  +[a+2;a:3]
8
````
Now it gives an error:
````
  +[a+2;a:3]
value error
a
^
parse error
````

**Comment by tavmem** [comment]
This was broken by the fix to issue #482 done on Sep 1, 2017.

**Comment by bakul** [comment]
Thanks!
While this specific issue is fixed, looks the root cause is not.
````
  (a;a:2)
value error
a
^
parse error
````
It should return `2 2`
